pkg/analyze/data_source.go

```diff
@@ -0,0 +1,78 @@
+package analyze
+
+import (
+	scyllav1listers "github.com/scylladb/scylla-operator/pkg/client/scylla/listers/scylla/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"reflect"
+)
+
+type DataSource interface {
+	ServiceLister() corev1listers.ServiceLister
+	SecretLister() corev1listers.SecretLister
+	ConfigMapLister() corev1listers.ConfigMapLister
+	ServiceAccountLister() corev1listers.ServiceAccountLister
+	ScyllaClusterLister() scyllav1listers.ScyllaClusterLister
+
+	AddShutdownBehaviour(shutdownBehaviour func())
+	Shutdown()
+}
```

Comment:
This would be much simpler as a struct with listers as attributes. You can have dedicated constructors: think `NewDataSourceFromClients` and `NewDataSourceFromArchive`.

Reply:
done